### PR TITLE
ROC-2759:  Refactor out domain models and sample data

### PR DESCRIPTION
**Pull request by sabahirfan**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/ROC-2759

### Change description ###
This PR refactors out domain/model classes in separate sub-module, sample data classes in separate sub-module. 
**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: cmc-domain/build.gradle [hunk 1]
```diff
@@ -0,0 +1,32 @@
+group 'uk.gov.hmcts.cmc'
+version '1.1.0'
+
+apply plugin: 'java'
+apply plugin: 'idea'
```
Comment (timja-kainos) [inline cmc-domain/build.gradle]: do we need this?

File: build.gradle [hunk 2]
```diff
@@ -78,53 +78,55 @@ configurations {
 
 dependencies {
 
-    compile project(':email-client')
-    compile group: 'org.springframework.boot', name: 'spring-boot-starter-web', version: versions.springBoot
-    compile group: 'org.springframework.boot', name: 'spring-boot-starter-actuator', version: versions.springBoot
-    compile group: 'org.springframework.boot', name: 'spring-boot-starter-jdbc', version: versions.springBoot
-    compile group: 'org.springframework.boot', name: 'spring-boot-starter-aop', version: versions.springBoot
+  compile project(':email-client')
```
Comment (timja-kainos) [inline build.gradle]: better to do a separate PR for this sort of indenting, if this PR drags at all you have a high risk of conflicts if anyone touches this file at all

File: ccd-adapter/build.gradle [hunk 3]
```diff
@@ -0,0 +1,14 @@
+group 'uk.gov.hmcts.cmc'
+version '1.1.0'
```
Comment (timja-kainos) [inline ccd-adapter/build.gradle]: we should be able to extract the version somehow, I assume this would never be different to the parent?

File: cmc-domain/src/test/resources/claim-application.json [hunk 4]
```diff
@@ -0,0 +1,74 @@
+{
```
Comment (timja-kainos) [inline cmc-domain/src/test/resources/claim-application.json]: any reason for this in domain? its for testing the api, so I would have though either an API module or in the core module

Comment (sabahirfan) [reply cmc-domain/src/test/resources/claim-application.json]: This is required by `ClaimDataSerializationTest`

File: ccd-adapter/build.gradle (hunk 3 above)
Comment (rlewan) [reply ccd-adapter/build.gradle]: It's definitely possible to use parent's version, I'm quite confident we've already done that.

File: cmc-domain-sample-data/build.gradle [hunk 5]
```diff
@@ -0,0 +1,16 @@
+group 'uk.gov.hmcts.cmc'
```
Comment (rlewan) [inline cmc-domain-sample-data/build.gradle]: I don't think there's a need to be that granular, sample data is perfectly fine to live along the domain. It's even better because it's easier to see you need to update sample data builders when changing the model.

File: cmc-domain/build.gradle [hunk 6]
```diff
@@ -0,0 +1,31 @@
+group 'uk.gov.hmcts.cmc'
```
Comment (rlewan) [inline cmc-domain/build.gradle]: `cmc-domain-model`?

File: cmc-domain/src/test/java/uk/gov/hmcts/cmc/claimstore/utils/PartyUtilsTest.java [hunk 7]
```diff
@@ -17,22 +18,20 @@
...
 import java.util.Optional;
 
-import static org.assertj.core.api.Assertions.assertThat;
-
 public class PartyUtilsTest {
 
     @Test
     public void getTypeMapsPartyTypesCorrectly() {
-        assertThat(PartyUtils.getType(SampleParty.builder().individual()))
+        Assertions.assertThat(PartyUtils.getType(SampleParty.builder().individual()))
```
Comment (rlewan) [inline cmc-domain/src/test/java/uk/gov/hmcts/cmc/claimstore/utils/PartyUtilsTest.java]: Use a static import please -- here and in all other places.

File: cmc-domain-sample-data/build.gradle (hunk 5 above)
Comment (sabahirfan) [reply cmc-domain-sample-data/build.gradle]: It was previously part of `test/java` not the `main/java`. The reason I pulled out is, it is required in domain-model tests and other tests (services & controllers in parent module) and we can add testCompile dependency. I prefer not to use name Sample***.java for Builder classes. If these are builder classes only, these should have proper names. Probably another cleanup on these.

File: cmc-domain/build.gradle (hunk 6 above)
Comment (sabahirfan) [reply cmc-domain/build.gradle]: seems better name.

File: ccd-adapter/build.gradle (hunk 3 above)
Comment (sabahirfan) [reply ccd-adapter/build.gradle]: Removed as were added by idea. Defaults would be from parent gradle.

File: cmc-domain/src/test/java/uk/gov/hmcts/cmc/claimstore/utils/PartyUtilsTest.java (hunk 7 above)
Comment (sabahirfan) [reply cmc-domain/src/test/java/uk/gov/hmcts/cmc/claimstore/utils/PartyUtilsTest.java]: Done

File: cmc-domain/build.gradle (hunk 1 above)
Comment (sabahirfan) [reply cmc-domain/build.gradle]: Removed

File: cmc-domain-model/build.gradle [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+]
+
+dependencies {
+  compile group: 'org.apache.commons', name: 'commons-lang3', version: '3.6'
+  compile group: 'cz.jirutka.validator', name: 'validator-collection', version: '2.2.0'
+  compile group: 'com.fasterxml.jackson.core', name: 'jackson-databind', version: versions.jackson
+  compile group: 'com.fasterxml.jackson.module', name: 'jackson-module-parameter-names', version: versions.jackson
+  compile group: 'com.fasterxml.jackson.datatype', name: 'jackson-datatype-jdk8', version: versions.jackson
+  compile group: 'com.fasterxml.jackson.datatype', name: 'jackson-datatype-jsr310', version: versions.jackson
+  compile group: 'cz.jirutka.validator', name: 'validator-collection', version: '2.2.0'
```
Comment (geek-soft) [inline cmc-domain-model/build.gradle]: Please remove duplicate.

File: cmc-domain-model/src/main/java/uk/gov/hmcts/cmc/claimstore/constraints/AgeRangeConstraintValidator.java [hunk 9]
```diff
@@ -0,0 +1,34 @@
+package uk.gov.hmcts.cmc.claimstore.constraints;
```
Comment (geek-soft) [inline cmc-domain-model/src/main/java/uk/gov/hmcts/cmc/claimstore/constraints/AgeRangeConstraintValidator.java]: This is claim data model so `claimstore` does not match here. It should be something like `uk.gov.hmcts.cmc.case.constraints`. I am happy for you to do it in separate PR if you prefer it.

File: cmc-domain-sample-data/build.gradle [hunk 10]
```diff
@@ -0,0 +1,13 @@
...
+sourceCompatibility = 1.8
+
+repositories {
+  mavenCentral()
+}
+
+dependencies {
+  compile project(':cmc-domain-model')
+
+  testCompile group: 'junit', name: 'junit', version: '4.12'
```
Comment (geek-soft) [inline cmc-domain-sample-data/build.gradle]: Is this needed for sample data project? Are we testing sample data somehow?

File: cmc-domain-model/build.gradle (hunk 8 above)
Comment (sabahirfan) [reply cmc-domain-model/build.gradle]: Removed.

File: cmc-domain-model/src/main/java/uk/gov/hmcts/cmc/claimstore/constraints/AgeRangeConstraintValidator.java (hunk 9 above)
Comment (sabahirfan) [reply cmc-domain-model/src/main/java/uk/gov/hmcts/cmc/claimstore/constraints/AgeRangeConstraintValidator.java]: I agree, and will do in a separate PR.

File: cmc-domain-sample-data/build.gradle (hunk 10 above)
Comment (sabahirfan) [reply cmc-domain-sample-data/build.gradle]: Not required. Removed.

File: cmc-domain-sample-data/build.gradle (hunk 5 above)
Comment (timja-kainos) [reply cmc-domain-sample-data/build.gradle]: they aren't builder classes, they are sample's because they provide defaults for tests
